api/routes/token.go

```diff
@@ -0,0 +1,118 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/shellhub-io/shellhub/api/apicontext"
+	"github.com/shellhub-io/shellhub/api/token"
+	"github.com/shellhub-io/shellhub/pkg/models"
+)
+
+const (
+	ListTokensURL  = "/token"
+	GetTokenURL    = "/token/:namespace"
+	CreateTokenURL = "/token/:namespace"
+	DeleteTokenURL = "/token/:namespace"
+	SetReadOnlyURL = "/token/:namespace"
+)
+
+func ListTokens(c apicontext.Context) error {
+	svc := token.NewService(c.Store())
+
+	query := filterQuery{}
+	if err := c.Bind(&query); err != nil {
+		return err
+	}
+
+	tokens, err := svc.ListTokens(c.Ctx(), query.Filter)
+	if err != nil {
+		return err
+	}
+
+	return c.JSON(http.StatusOK, tokens)
+}
+
+func CreateToken(c apicontext.Context) error {
+	svc := token.NewService(c.Store())
+
+	var token models.Token
+	if err := c.Bind(&token); err != nil {
+		return err
+	}
+
+	namespace := ""
+	if v := c.Username(); v != nil {
+		namespace = v.Username
+	}
+
+	if _, err := svc.CreateToken(c.Ctx(), namespace); err != nil {
+		if err == token.ErrUnauthorized {
```

Comment:
To mach here, as well as other `err` special cases, it is better to use `switch` as it allows for same construction for one or more special cases and the explicit default one.